junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/config/DisplayNameGeneratorParameterConverter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.config;
+
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.ReflectionUtils;
+import org.junit.platform.engine.ConfigurationParameters;
+
+/**
+ * @since 5.5
+ */
+class DisplayNameGeneratorParameterConverter {
+
+	private static final Logger logger = LoggerFactory.getLogger(DisplayNameGeneratorParameterConverter.class);
+
+	DisplayNameGenerator get(ConfigurationParameters configurationParameters, String key) {
+		return configurationParameters.get(key).map(String::trim).filter(
+			className -> !className.isEmpty()).<Class<? extends DisplayNameGenerator>> map(
+				this::getClassFrom).<DisplayNameGenerator> map(clazz -> ReflectionUtils.newInstance(clazz)).orElse(
+					new DisplayNameGenerator.Standard());
+	}
+
+	@SuppressWarnings({ "unchecked" })
+	private Class<? extends DisplayNameGenerator> getClassFrom(String className) {
+		Class<?> clazz = ReflectionUtils.tryToLoadClass(className).toOptional().orElse(null);
```

Comment:
The whole point of using `Try` is so that the failure information does not get lost.

If we silently swallow the failure (i.e., exception stack trace), the user will not know why it failed.

Please rework this so that any exception that occurs while trying to load the class is reported along with the log statement.

If you need inspiration, check out other usages of `Try` within the code base.

Reply:
Updated PR with the new style (consuming exception by try).
Please have a look.